conformance/lang/expressions/error_constructor.balt

```diff
@@ -0,0 +1,700 @@
+
+type E3 error<record {|string message; boolean fatal?; string...;|}>;
+
+function init() {
+    error err1 = error E1("error message 1", fatal = false, msg = "msg");
+    io:println(err1.detail()); // @output {"fatal":false,"msg":"msg"}
+
+    error err2 = error E2("error message 2", fatal = false, msg = "msg");
+    io:println(err2.detail()); // @output {"fatal":false,"msg":"msg"}
+
+    error err3 = error E3("error message 3", fatal = false, message = "msg");
+    io:println(err3.detail()); // @output {"message":"msg","fatal":false}
+
+    error err = error("error message");
+    error err4 = error E1("error message 4", err, fatal = false, msg = "msg");
+    io:println(err4.detail()); // @output {"fatal":false,"msg":"msg"}
+
+    error err5 = error E2("error message 5", err, fatal = false, msg = "msg");
+    io:println(err5.detail()); // @output {"fatal":false,"msg":"msg"}
+
+    error err6 = error E3("error message 6", err, fatal = false, message = "msg");
+    io:println(err6.detail()); // @output {"message":"msg","fatal":false}
+}
+
+Test-Case: output
+Description: Test allowing named-arg when the type descriptor D of the detail mapping does not have any individual fields.
+Labels: error-constructor-expr, error-type, error-type-parameter, error:detail, method-call-expr, module-type-defn,
+        record-type, string
+
+type E1 error<record {}>;
+
+type E2 error<map<string>>;
+
+function init() {
+    error err1 = error("error message 1", fatal = false, msg = "msg");
+    io:println(err1.detail()); // @output {"fatal":false,"msg":"msg"}
+
+    error err2 = error E1("error message 2", fatal = false, msg = "msg");
+    io:println(err2.detail()); // @output {"fatal":false,"msg":"msg"}
+
+    error err3 = error E2("error message 3", fatal = "false", msg = "msg");
+    io:println(err3.detail()); // @output {"fatal":"false","msg":"msg"}
+
+    error err = error("error message");
+    error err4 = error("error message 1", err, fatal = false, msg = "msg");
+    io:println(err4.detail()); // @output {"fatal":false,"msg":"msg"}
+
+    error err5 = error E1("error message 2", err, fatal = false, msg = "msg");
+    io:println(err5.detail()); // @output {"fatal":false,"msg":"msg"}
+
+    error err6 = error E2("error message 3", err, fatal = "false", msg = "msg");
+    io:println(err6.detail()); // @output {"fatal":"false","msg":"msg"}
+}
+
+Test-Case: output
+Description: Test fields with default values will also be added to the detail record in the same way as
+             the mapping-constructor-expr adds fields with default values based on the contextually expected type.
+Labels: error-constructor-expr, error-type, error-type-parameter, module-type-defn, record-type, string
+
+type R1 record {
+    string errorCode;
+    string message = "Something's wrong";
+    boolean fatal = false;
+};
+
+type R2 record {|
+    string errorCode;
+    string message = "Something's wrong";
+    boolean fatal = false;
+|};
```

Comment:
Let's add tests verifying calling  `ImmutableClone` for cases where default values are used too.

Reply:
A test added at L280.